modules/vector-sets/hnsw.c

```diff
@@ -390,10 +390,12 @@ uint32_t random_level(void) {
 }
 
 /* Create new HNSW index, quantized or not. */
-HNSW *hnsw_new(uint32_t vector_dim, uint32_t quant_type, uint32_t m) {
+HNSW *hnsw_new(uint32_t vector_dim, uint32_t quant_type, uint32_t m, uint8_t n_threads) {
     HNSW *index = hmalloc(sizeof(HNSW));
     if (!index) return NULL;
 
+    assert(n_threads <= HNSW_MAX_THREADS); // Change to if > -> = MAX or return NULL ?
```

Comment:
I suggest just trimming this. If > HNSW_MAX_THREADS just set it to HNSW_MAX_THREADS. Too strong failure mode for something that still allows the system to work IMHO.

Reply:
Agreed, I'll cap it to HNSW_MAX_THREADS